src/main/java/com/google/api/codegen/discovery/viewmodel/SampleView.java

```diff
@@ -146,6 +148,17 @@ public String resourceRoot() {
   @Nullable
   public abstract List<String> methodParamAssigments();
 
+  // JS specific section...
+  @Nullable
+  public abstract String paramsVarName();
+
+  /**
+   * Returns true if the page streaming call requires a `params` object to set the `pageToken`
+   * field, even if there are no required params.
```

Comment:
Sorry, that's not what I meant ... by the usual case, I mean the case where there are required parameters. The doc string at https://github.com/saicheems/toolkit/blob/1bb05a49bf12e5f474cc284837c3b5069291a7a1/src/main/java/com/google/api/codegen/discovery/transformer/js/JSSampleMethodToViewTransformer.java#L99 is clear about this point ("If there are fields, or ...").